packages/sdk-codegen/src/kotlin.gen.ts

```diff
@@ -404,27 +410,31 @@ ${props.join(this.propDelimiter)}
   }
 
   httpCall(indent: string, method: IMethod) {
+    const bump = indent + this.indentStr
+    const methodName = this.it(method.httpMethod.toLowerCase())
     const request = this.useRequest(method) ? 'request.' : ''
-    const type = this.typeMap(method.type)
+    const tSuccess = this.typeMap(method.type).name
     const args = this.httpArgs(indent, method)
+    const path = this.httpPath(method.endpoint, request)
+    const functionArgs = [path].concat(args)
     // TODO don't currently need these for Kotlin
     // const errors = this.errorResponses(indent, method)
-    return `${indent}return ${this.it(method.httpMethod.toLowerCase())}<${
-      type.name
-    }>(${this.httpPath(method.endpoint, request)}${args ? ', ' + args : ''})`
+    return `${indent}return ${methodName}<${tSuccess}, ${this.tFailure}>(
+${bump}${functionArgs.join(', ')}
+${indent})`
   }
 
   streamCall(indent: string, method: IMethod) {
     const request = this.useRequest(method) ? 'request.' : ''
+    const methodName = this.it(method.httpMethod.toLowerCase())
+    const tSuccess = 'ByteArray'
```

Comment:
For streams, we always return ByteArray for TSuccess.  I kept `com.looker.sdk.Error` for TFailure, but maybe that should be ByteArray as well?  Or maybe the streaming methods should also do deserialization?  I don't know much about how we're handling streaming methods in the other SDKs, so I tried to change as little here as possible.

Reply:
The kotlin "streaming" support was a total hack just to support binary payload downloading. Same is true for the Swift SDK. We still need real streaming functionality.